src/taker/offers.rs

```diff
@@ -153,12 +140,69 @@ pub async fn sync_offerbook_with_addresses(
 }
 
 /// Retrieves advertised maker addresses from directory servers based on the specified network.
-pub async fn get_advertised_maker_addresses(
-    network: Network,
+pub async fn fetch_addresses_from_dns(
+    socks_port: Option<u16>,
+    directory_server_address: String,
+    _network: Network,
 ) -> Result<Vec<MakerAddress>, DirectoryServerError> {
-    Ok(if network == Network::Regtest {
-        get_regtest_maker_addresses()
-    } else {
-        sync_maker_addresses_from_directory_servers(network).await?
-    })
+    let max_retries = 10;
+    let mut retries = 0;
+    loop {
+        let result: Result<Vec<MakerAddress>, DirectoryServerError> = (async {
+            let mut stream: TcpStream = Socks5Stream::connect(
+                format!("127.0.0.1:{}", socks_port.unwrap_or(19050)).as_str(),
+                directory_server_address.as_str(),
+            )
+            .await
+            .map_err(|_e| {
+                DirectoryServerError::Other(
+                    "Issue with fetching maker address from directory server",
+                )
+            })?
+            .into_inner();
+
+            let request_line = "GET\n";
+            stream
+                .write_all(request_line.as_bytes())
+                .await
+                .map_err(|_e| DirectoryServerError::Other("Error sending the request"))?;
+
+            let mut response = String::new();
+            stream
+                .read_to_string(&mut response)
+                .await
+                .map_err(|_e| DirectoryServerError::Other("Error receiving the response"))?;
+
+            log::warn!("Received: {}", response);
+
+            let addresses: Vec<MakerAddress> = response
+                .lines()
+                .map(|addr| MakerAddress::new(addr.to_string()))
+                .collect();
+
+            Ok(addresses)
+        })
+        .await;
+
+        match result {
+            Ok(addresses) => {
+                if cfg!(feature = "integration-test") && addresses.len() < 2 {
+                    thread::sleep(Duration::from_secs(10));
+                    continue;
+                }
+                if retries < max_retries {
+                    retries += 1;
+                    thread::sleep(Duration::from_secs(10));
+                    continue;
+                }
```

Comment:
The effect of this code is the taker will keep pinging the DNS no matter how many addresses it already has. This slows down the test and also will be very awkward for a real taker. 

Also this code is saving our day because the above "is there 2 makers" conditional isn't gonna work. Some tests require 3 makers. Because we keep fetching for a long time so all goes well.

For now just move this section into the cfg!() if clause. Later on we can figure something more cleaner.